kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py

```diff
@@ -26,13 +26,16 @@ def __init__(self, name, init_config, instances):
         self.config = KafkaConfig(self.init_config, self.instance)
         self._context_limit = self.config._context_limit
         tls_context = self.get_tls_context()
-        self.client = make_client(self, self.config, tls_context)
+        self.client = make_client(self.config, tls_context, self.log)
 
     def check(self, _):
         """The main entrypoint of the check."""
         # Fetch Kafka consumer offsets
         self.client.reset_offsets()
 
+        consumer_offsets = {}
```

Comment:
Not too sure if this is the best approach, but because I wanted to move `self.client.should_get_highwater_offsets()` back to the check, we needed to have the `consumer_offsets` already initialized here. The other option is to set `self._consumer_offsets` and `self._highwater_offsets` to `{}` here instead, but I was thinking it may be "safer" to use few instance variables?